homeassistant/components/matter/cover.py

```diff
@@ -0,0 +1,157 @@
+"""Matter cover."""
+from __future__ import annotations
+
+from typing import Any
+
+from chip.clusters import Objects as clusters
+from matter_server.client.models import device_types
+
+from homeassistant.components.cover import (
+    ATTR_POSITION,
+    CoverEntity,
+    CoverEntityDescription,
+    CoverEntityFeature,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import Platform
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import LOGGER
+from .entity import MatterEntity
+from .helpers import get_matter
+from .models import MatterDiscoverySchema
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up Matter cover from Config Entry."""
+    matter = get_matter(hass)
+    matter.register_platform_handler(Platform.COVER, async_add_entities)
+
+
+class MatterCover(MatterEntity, CoverEntity):
+    """Representation of a Matter cover."""
+
+    entity_description: CoverEntityDescription
+
+    @property
+    def supported_features(self) -> CoverEntityFeature:
+        """Flag supported features."""
+        features = (
+            CoverEntityFeature.OPEN
+            | CoverEntityFeature.CLOSE
+            | CoverEntityFeature.STOP
+            | CoverEntityFeature.SET_POSITION
+        )
+        return features
+
+    @property
+    def current_cover_position(self) -> int:
+        """Return the current position of cover."""
+        if self._attr_current_cover_position:
+            current_position = self._attr_current_cover_position
+        else:
+            current_position = self.get_matter_attribute_value(
+                clusters.WindowCovering.Attributes.CurrentPositionLiftPercentage
+            )
+
+        assert current_position is not None
+
+        LOGGER.info(
```

Comment:
Please don't do side effects like logging in entity state properties. We can remove this.